Daily/main crashing on build

Storage migration attempts log, resulting in `kotlin.UninitializedPropertyAccessException: lateinit property logger has not been initialized`

`
2025-08-11 09:54:57.367 28557-28557 AndroidRuntime          pid-28557                            E  	at org.koin.core.resolution.CoreResolver.resolveFromContextOrNull(CoreResolver.kt:74) (Ask Gemini)
    at org.koin.core.resolution.CoreResolver.resolveFromContextOrNull$default(CoreResolver.kt:72)
    at org.koin.core.resolution.CoreResolver.resolveFromContext(CoreResolver.kt:69)
    at org.koin.core.scope.Scope.resolveFromContext(Scope.kt:319)
    at org.koin.core.scope.Scope.stackParametersCall(Scope.kt:282)
    at org.koin.core.scope.Scope.resolveInstance(Scope.kt:268)
    at org.koin.core.scope.Scope.resolve(Scope.kt:241)
    at org.koin.core.scope.Scope.get(Scope.kt:223)
    at com.fsck.k9.preferences.KoinModuleKt.preferencesModule$lambda$25$lambda$10(KoinModule.kt:142)
    at com.fsck.k9.preferences.KoinModuleKt.$r8$lambda$MyjXYMuUkYRfpmOWlnRvJc7tqZE(KoinModule.kt:0)
    at com.fsck.k9.preferences.KoinModuleKt$$ExternalSyntheticLambda2.invoke(R8$$SyntheticClass:0)
    at org.koin.core.instance.InstanceFactory.create(InstanceFactory.kt:49)
    ... 36 more
  Caused by: kotlin.UninitializedPropertyAccessException: lateinit property logger has not been initialized
    at net.thunderbird.core.logging.legacy.Log.getLogger(Log.kt:32)
    at net.thunderbird.core.logging.legacy.Log.d(Log.kt:108)
    at com.fsck.k9.preferences.migration.DefaultStorageMigrationHelper.readValue(DefaultStorageMigrationHelper.kt:34)
    at com.fsck.k9.preferences.migration.StorageMigrationTo28.ensureAvatarSet(StorageMigrationTo28.kt:16)
    at com.fsck.k9.preferences.migration.StorageMigrations.upgradeDatabase(StorageMigrations.kt:37)
    at com.fsck.k9.preferences.K9StoragePersister.openDB(K9StoragePersister.java:54)
    at com.fsck.k9.preferences.K9StoragePersister.loadValues(K9StoragePersister.java:154)
    at com.fsck.k9.Preferences.getStorage(Preferences.kt:60)
    at com.fsck.k9.preferences.KoinModuleKt.preferencesModule$lambda$25$lambda$4(KoinModule.kt:44)
    at com.fsck.k9.preferences.KoinModuleKt.$r8$lambda$qbHAFINX2GnWHb4YxS6-YdF0cxA(KoinModule.kt:0)
    at com.fsck.k9.preferences.KoinModuleKt$$ExternalSyntheticLambda20.invoke(R8$$SyntheticClass:0)
    at org.koin.core.instance.InstanceFactory.create(InstanceFactory.kt:49)
    ... 56 more
`

I think this is caused by the first migration code done since the new logging system was integrated. We need the logging initialized before a migration can occur.

--- a/app-common/src/main/kotlin/net/thunderbird/app/common/BaseApplication.kt
+++ b/app-common/src/main/kotlin/net/thunderbird/app/common/BaseApplication.kt
@@ -51,16 +51,16 @@ abstract class BaseApplication : Application(), WorkManagerConfiguration.Provide
     override fun attachBaseContext(base: Context?) {
         Core.earlyInit()
 
-        super.attachBaseContext(base)
-
         // Start Koin early so it is ready by the time content providers are initialized.
         DI.start(this, listOf(provideAppModule()))
+        Log.logger = logger
+
+        super.attachBaseContext(base)
     }
 
     override fun onCreate() {
         super.onCreate()
 
-        Log.logger = logger
         K9.init(this)
         Core.init(this)
         initializeAppLanguage()

